Description: Tighten permissions of fbguard lock file
 Creating a worls-writable file is a bad idea, even if it currently lives
 in a directory restricted for read/write/use to firebird:firebird
Bug-Debian: https://bugs.debian.org/767497
--- a/src/utilities/guard/guard.cpp
+++ b/src/utilities/guard/guard.cpp
@@ -146,7 +146,7 @@ int CLIB_ROUTINE main( int argc, char **
 	}
 
 	// get and set the umask for the current process
-	const ULONG new_mask = 0000;
+	const ULONG new_mask = 0007;
 	const ULONG old_mask = umask(new_mask);
 
 	// exclusive lock the file
--- a/src/utilities/guard/util.cpp
+++ b/src/utilities/guard/util.cpp
@@ -253,7 +253,7 @@ int UTIL_ex_lock(const TEXT* file)
 	Firebird::PathName expanded_filename = fb_utils::getPrefix(fb_utils::FB_DIR_GUARD, file);
 
 	// file fd for the opened and locked file
-	int fd_file = open(expanded_filename.c_str(), O_RDWR | O_CREAT, 0666);
+	int fd_file = open(expanded_filename.c_str(), O_RDWR | O_CREAT, 0660);
 	if (fd_file == -1)
 	{
 		fprintf(stderr, "Could not open %s for write\n", expanded_filename.c_str());
